### PR TITLE
csvkit: fix build

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

#97479 cc @NixOS/nixos-release-managers .

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
sorry, I saw the other first, I think a fix already went it

for the commit message, please have it adhere to CONTRIBUTing:

```
csvkit: <message>
```

ZHF is just a period of time

**Comment by doronbehar** [comment]
> sorry, I saw the other first, I think a fix already went it

I saw the merge conflicts, and hence the changes performed by @zookatron at https://github.com/NixOS/nixpkgs/commit/32c9ee2cfc18cf9d798011b3822c176a2ce819d9 . TBH, I think my changes are better. I rebased it all upon the current master. All of the packages used in my version are of their latest versions, and there are no overrides at all used. An upstream PR was fetched so all tests should pass, plus a version bump.

> for the commit message, please have it adhere to CONTRIBUTing:

I do adhere to it. The PR includes 4 commits each dealing with a different package and the PR title is just the name of the branch.

**Comment by zookatron** [comment]
Thanks for taking the time to update these packages more thoroughly @doronbehar. For my changes I was just trying to make the minimal changes necessary for [Zero Hydra Failures](https://github.com/NixOS/nixpkgs/issues/97479) and I did not update the package version itself as my understanding is that we should avoid updating the package versions themselves when back-porting changes to an already stable branch.

File: pkgs/tools/text/csvkit/default.nix [hunk 1]
```diff
@@ -1,44 +1,33 @@
...
-  version = "1.0.4";
+  version = "1.0.5";
 
   src = python3.pkgs.fetchPypi {
     inherit pname version;
-    sha256 = "1830lb95rh1iyi3drlwxzb6y3pqkii0qiyzd40c1kvhvaf1s6lqk";
+    sha256 = "1ffmbzk4rxnl1yhqfl58v7kvl5m9cbvjm8v7xp4mvr00sgs91lvv";
   };
 
+  patches = [
```
Comment (jonringer) [inline pkgs/tools/text/csvkit/default.nix:12]: please add a comment as to why the patch is necesary, and when it's likely to be removed

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -1309,6 +1309,8 @@ in {
 
   cram = callPackage ../development/python-modules/cram { };
 
+  crate = callPackage ../development/python-modules/crate { };
+
   crashtest = callPackage ../development/python-modules/crashtest { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:1314]: t comes after s
```suggestion
  crashtest = callPackage ../development/python-modules/crashtest { };

  crate = callPackage ../development/python-modules/crate { };
```

File: pkgs/development/python-modules/crate/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  checkInputs = [
+    pytestCheckHook
+    sqlalchemy
+    mock
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = "http://www.sqlalchemy.org/";
+    description = "A Python SQL toolkit and Object Relational Mapper";
+    license = licenses.mit;
```
Comment (jonringer) [inline pkgs/development/python-modules/crate/default.nix:33]: I would really like to have a maintainer of this

Comment (doronbehar) [reply pkgs/development/python-modules/crate/default.nix:33]: Sure, myself. I also updated the whole meta attribute as this one was of a copy paste.

File: pkgs/tools/text/csvkit/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/tools/text/csvkit/default.nix:12]: :+1:.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/top-level/python-packages.nix]: Also fixed, now.

**Comment by doronbehar** [comment]
Are we still interested in this @jonringer ?

**Comment by jonringer** [comment]
I think for master, this is fine, but for release-20.09 it's adding a new package, and affecting some others.

I was on vacation and only had a laptop, so I couldn't do large rebuilds

File: pkgs/development/python-modules/crate/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+, urllib3
+, isPy3k
+, mock
+, sqlalchemy
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "crate";
+  version = "0.24.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/crate/default.nix:13]:
```suggestion
  version = "0.24.0";
  disabled = !isPy3k;
```
```
  Processing ./crate-0.24.0-py2.py3-none-any.whl
  ERROR: Package 'crate' requires a different Python: 2.7.18 not in '>=3.4'
```

Comment (doronbehar) [reply pkgs/development/python-modules/crate/default.nix:13]: Fixed in crate's init commit.

**Comment by jonringer** [comment]
actually, seeing as the package addition was to fix a different build, maybe it should be included

cc @worldofpeace

File: pkgs/development/python-modules/crate/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  checkInputs = [
+    pytestCheckHook
+    sqlalchemy
+    mock
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/crate/crate-python";
+    description = "A Python client library for CrateDB";
+    license = licenses.apsl20;
```
Comment (drewrisinger) [inline pkgs/development/python-modules/crate/default.nix:34]:
```suggestion
    license = licenses.asl20;
```

Comment (doronbehar) [reply pkgs/development/python-modules/crate/default.nix]: Thanks :). Fixed in crate's init commit.

**Review by drewrisinger — APPROVED** [review]
* [x] Diff LGTM
* [x] Commits LGTM
* [x] Builds

```
https://github.com/NixOS/nixpkgs/pull/97517
5 packages built:
csvkit python37Packages.agate-sql python37Packages.crate python38Packages.agate-sql python38Packages.crate
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 97517 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>5 packages built:</summary>  <ul>    <li>csvkit</li>    <li>python37Packages.agate-sql</li>    <li>python37Packages.crate</li>    <li>python38Packages.agate-sql</li>    <li>python38Packages.crate</li>  </ul></details>

**Pull request merged**